packages/core/core/src/resolveOptions.js

```diff
@@ -189,6 +189,9 @@ export default async function resolveOptions(
     shouldTrace: initialOptions.shouldTrace ?? false,
     cacheDir,
     watchDir,
+    baseWatcherOptions: initialOptions.baseWatcherOptions ?? {
```

Comment:
The reason I didn't go for `watcherOptions` is because I feel like it's somewhat misleading/suggestive precisely since there are other options (e.g. `watchDir`) that configure the watcher. Reading `base...` implies there may be others. Happy to change if you still feel `watcherOptions` is good enough.